drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/s3_zip_files_importer.rb

```diff
@@ -0,0 +1,71 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# This class iterates over the zip file objects in a bucket "directory" and
+# calls an importer on that directory after extraction.
+# FIXME: Does this already exist somewhere?
```

Comment:
We don't have anything that does that exactly.
`Importers::HmisAutoMigrate::S3` has much of this logic.  Not the best structure over there, but `recently_imported?` checks for having already processed the file.